dbt/adapters/bigquery/relation_configs/_options.py

```diff
@@ -21,6 +21,7 @@ class BigQueryOptionsConfig(BigQueryBaseRelationConfig):
     refresh_interval_minutes: Optional[float] = 30
     expiration_timestamp: Optional[datetime] = None
     max_staleness: Optional[str] = None
+    allow_non_incremental_definition: Optional[bool] = None
```

Comment:
The database defaults to `False`, so we should do the same here so that we don't recognize it as a change each time.